core/src/autogluon/core/utils/early_stopping.py

```diff
@@ -0,0 +1,103 @@
+
```

Comment:
Should we caveat these are strategies only being used by certain tree models, and the NN models use different early-stopping strategies?

Reply:
The plan is to only mention them on a model-by-model basis. Each model would implement support for it separately. We do want the NN models to eventually use these strategies however. Added mention of it in LightGBM docs.